### PR TITLE
[SWE-19] Create Account Backend

### PR DESCRIPTION
**Pull request by datderek**
Task ID: [SWE-19](https://linear.app/swe-dem-boys/issue/SWE-19/create-account-backend)

## PR Description

Created the backend to the Account creation page. Registers the user using AWS Amplify Auth. Redirects to `/login` on success.

### CODE REVIEW

**Comment by datderek** [comment]
Just an incomplete, inital implementation. Still need to address redirect and error messages from Cognito.

@jackfales @sammyoto Trying to figure out the best approach to parse formData from the front-end. Current implementation uses the formidable library to parse through the formData.

There are several other approaches:
1.

**Comment by datderek** [comment]
Setup Amplify following [Project Setup Tutorial](https://docs.amplify.aws/lib/project-setup/prereq/q/platform/js/). Added a lot of files, might need to ignore some additional files it created such as `.vscode`?

**Comment by datderek** [comment]
@jackfales Switched back to sending FormData to the backend as a JSON object as I couldn't find a way an easy way to parse the FormData object from the request body besides the alternatives mentioned above.

Discussed with @sammyoto that it may be best to use App Router that Nextjs 13.4 offers, but that will take some migration that I think should be done in a follow-up task.

File: pages/api/create-account.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { Auth } from 'aws-amplify'
+
+export default async function handler(req, res) {
+  if (req.method !== 'POST') return res.status(405).send();
+
+  const { username, password, lastName, firstName } = JSON.parse(req.body);
```
Comment (jackfales) [inline pages/api/create-account.js:6]: Beautiful

Comment (jackfales) [inline pages/api/create-account.js:6]: Beautiful

Comment (jackfales) [inline pages/api/create-account.js:6]: Beautiful

nit: Have some form of consistency when listing variables. Either alphabetical or by use-case. It makes it easier across the codebase to look for certain variables or add new ones in the correct location.

Alphabetical: `firstName, lastName, password, username`
Use-case: `username, password, firstName, lastName`

File: pages/create-account.js [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import Head from 'next/head';
 import React, { useState } from 'react';
+import { useRouter } from 'next/router';
 import styles from '../styles/Home.module.css';
 
 export default function Account() {
```
Comment (jackfales) [inline pages/create-account.js:6]: Rename to `CreateAccount`

**Review by jackfales — APPROVED** [review]
Great job, couple of nit comments but should be good to merge.

**Review by sammyoto — APPROVED** [review]
Looks good. Just make sure you can run the amplify commands on your end.

File: pages/api/create-account.js (hunk 1 above)
Comment (datderek) [reply pages/api/create-account.js]: Changed to Alphabetical for consistency

**Review by sammyoto — APPROVED** [review]
Tested everything on my end, looks good.

**Pull request merged**